librz/core/canalysis.c

```diff
@@ -5755,6 +5755,110 @@ RZ_IPI bool rz_core_analysis_var_rename(RzCore *core, const char *name, const ch
 	return true;
 }
 
+static bool is_in_data_section(RzCore *core, const ut64 address) {
+	if (address < 256 || address == UT64_MAX) {
+		return false;
+	}
+	RzBinObject *o = rz_bin_cur_object(core->bin);
+	const RzBinSection *section = o ? rz_bin_get_section_at(o, address, core->io->va) : NULL;
+	if (!section || section->size < 2) {
+		return false;
+	}
+
+	if (strstr(section->name, "data")) {
+		// this ensures that the section will always be used if
+		// contains `data` substring regardless of their permissions.
+		return true;
+	}
+
+	return section->perm && !(section->perm & RZ_PERM_X);
+}
```

Comment:
This will be replaced by the new function in https://github.com/rizinorg/rizin/pull/2878 PR